Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm

```diff
@@ -321,6 +321,34 @@
 
 #endif
 
+#if ENABLE(SCREEN_TIME)
+void PageClientImplCocoa::installScreenTimeWebpageController()
+{
+    [m_webView _installScreenTimeWebpageController];
+}
+
+void PageClientImplCocoa::updateScreenTimeWebpageControllerURL(WKWebView *webView)
+{
+    if (!PAL::isScreenTimeFrameworkAvailable())
+        return;
+
+    RetainPtr screenTimeWebpageController = [webView _screenTimeWebpageController];
+    if (!screenTimeWebpageController)
+        return;
+
+    NakedPtr<WebKit::WebPageProxy> pageProxy = [webView _page];
+    if (pageProxy && !pageProxy->preferences().screenTimeEnabled()) {
+        [webView _uninstallScreenTimeWebpageController];
+        return;
+    }
+
+    if (pageProxy)
+        pageProxy->makeViewBlankIfUnpaintedSinceLastLoadCommit();
```

Comment:
Also, we shouldn't use `makeViewBlankIfUnpaintedSinceLastLoadCommit` without making refactoring to that method. Currently, it is specifically for JS prompts, as it has logging about that and is also conditional based on a linked-on-or-after check.